### PR TITLE
Split `DuplicateLine` into `DuplicateLineUp` and `DuplicateLineDown`

### LINKED ISSUE #9601: Split `DuplicateLine` into `DuplicateLineUp` and `DuplicateLineDown`.
### Check for existing issues

- [X] Completed

### Describe the feature

Here is the current default keymap code for `DuplicateLine`:

```json
"alt-shift-up": [
  "editor::DuplicateLine",
  {
    "move_upwards": true
  }
],
"alt-shift-down": "editor::DuplicateLine",
```

As it is, duplicating a line upwards isn't discoverable via the command palette, since the functionality is governed by a bool within `DuplicateLine`. We try to match VS Code's keymap by default; VS Code has separate actions for these (`Copy Line Up` and `Copy Line Down`).  This would be a breaking change, but we have been accepting breaking changes to make the keymap more consistent and polished.

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by xyzqm**
Fixes #9601

Release Notes:
- `DuplicateLine` is now split into `DuplicateLineUp` and `DuplicateLineDown`

### CODE REVIEW

File: crates/editor/src/editor_tests.rs [hunk 1]
```diff
@@ -3117,7 +3117,7 @@ fn test_duplicate_line(cx: &mut TestAppContext) {
                 DisplayPoint::new(3, 0)..DisplayPoint::new(3, 0),
             ])
         });
-        view.duplicate_line(&DuplicateLine::default(), cx);
+        view.duplicate_line_down(&DuplicateLineDown::default(), cx);
```
Comment (SomeoneToIgnore) [inline crates/editor/src/editor_tests.rs:3120]: NIT: do you really need those `::default()` now given that both `Duplicate*` are field-less structs now?

**Review by SomeoneToIgnore — APPROVED** [review]
LGTM but needs a rebase.

This obviously breaks things for existing configs, but given that the issue was created by @\JosephTLyons , most probably that is the plan.

@mrnugget do you have any extra concerns related to compatibility or other things here?

**Comment by danielzsh** [comment]
> Yeah, can we create a serde alias or something for the default? As far as I know these shortcuts are used often and I don't think we should break them.

How can I create such an alias? I found `#[serde(alias = ...)]`, but I'm not sure where to put it 😓

**Comment by mrnugget** [comment]
I'm actually not sure myself. I think @maxbrunsfeld proposed using a serde alias in a different PR for backwards compatibility, but I'm not sure how to use it for struct names here.

This is the code where we turn a string into an action:

https://github.com/zed-industries/zed/blob/ff685b299db35890c466382b36a09d2e311f5965/crates/settings/src/keymap_file.rs#L78

It seems like we'd have to introduce aliases at this level.

Not sure whether that's worth it, though. Maybe it is okay to break compatibility then, since it would "only" break the customized configurations, right?

**Comment by SomeoneToIgnore** [comment]
So far, we broke keybindings a few times drastically, all in VSCode compatibility favor, I also do not have much to oppose to "why not once more in that direction" idea.

So let's merge it and consider raising a compatibility discussion on the next PR with the similar config changes.

**Pull request merged**